conferences/ticketing_storage.go

```diff
@@ -0,0 +1,605 @@
+	}
+
+	return nil
+}
+
+// createSlotClaim saves a slot claim and returns it with the populated ID
+func createSlotClaim(ctx context.Context, tx *sqldb.Tx, slotClaim *SlotClaim) (*SlotClaim, error) {
+	var err error
+	// FIXME: Add a check for capacity not exceeded on conference.
+	for i := 0; i < 3; i++ {
+		row := queryRow(ctx, tx,
+			`INSERT INTO slot_claim (ticket_id, redeemed, conference_slot_id) VALUES ($1, $2, $3)
+		RETURNING id, ticket_id, redeemed`,
+			slotClaim.TicketID, slotClaim.Redeemed, slotClaim.ConferenceSlot.ID)
+
+		results := SlotClaim{}
+		err = row.Scan(&results.ID, &results.TicketID, &results.Redeemed)
+
+		if err != nil {
+			return nil, fmt.Errorf("saving slot claim: %w", err)
+		}
+
+		results.ConferenceSlot = slotClaim.ConferenceSlot
+		return &results, nil
+	}
+	return nil, fmt.Errorf("failed to insert: %w", err)
+}
+
+const (
+	tableAttendee               = "attendee"
+	tableAttendeeSlotClaims     = "attendee_to_slot_claims"
+	slotClaimIDUniqueConstraint = "slot_claim_id_is_unique"
+)
+
+// updateAttendee saves the passed attendee attributes on top of the existing one.
+func updateAttendee(ctx context.Context, tx *sqldb.Tx, attendee *Attendee) (*Attendee, error) {
+	res, err := exec(ctx, tx,
+		`UPDATE attendee SET email = $1, coc_accepted = $2 WHERE id = $3`,
+		attendee.Email, attendee.CoCAccepted, attendee.ID)
+
+	if err != nil {
+		return nil, fmt.Errorf("updating attendee: %w", err)
+	}
+
+	ra, err := res.RowsAffected()
+	if err != nil {
+		return nil, fmt.Errorf("updating attendee: %w", err)
+	}
+	if ra == 0 {
+		return nil, fmt.Errorf("attendee was not updated")
+	}
+
+	for i := range attendee.Claims {
+		c := attendee.Claims[i]
+		res, err = exec(ctx, tx,
+			`INSERT INTO attendee_to_slot_claims (attendee_id, slot_claim_id) 
+			VALUES ($1, $2) 
+		ON CONFLICT ON CONSTRAINT slot_claim_id_is_unique DO UPDATE SET attendee_id = $3`,
+			attendee.ID, c.ID, attendee.ID)
+		if err != nil {
+			return nil, fmt.Errorf("inserting attendee claims: %w", err)
+		}
+		ra, err := res.RowsAffected()
+		if err != nil {
+			return nil, fmt.Errorf("inserting attendee claims: %w", err)
+		}
+		if ra == 0 {
+			return nil, fmt.Errorf("could not create claim")
+		}
+	}
+
+	return attendee, nil
+}
+
+const (
+	tableClaimPayment                = "claim_payment"
+	tableFinancialInstrumentMoney    = "payment_method_money"
+	tableMoneyToPayment              = "payment_method_money_to_claim_payment"
+	tableFinancialInstrumentDiscount = "payment_method_conference_slotdiscount"
+	tableDiscountToPayment           = "payment_method_conference_slotdiscount_to_claim_payment"
+	tableFinancialInstrumentCredit   = "payment_method_credit_note"
+	tableCreditToPayment             = "payment_method_credit_note_to_claim_payment"
+)
+
+func insertMoneyPayment(ctx context.Context, tx *sqldb.Tx, claimPaymentID uint64, payment *PaymentMethodMoney) (*PaymentMethodMoney, error) {
+	if payment.ID != 0 { // BIGSERIAL starts in 1
+		return payment, nil
+	}
```

Comment:
What is the purpose of this? It seems to me this is conditionally inserting a payment. Wouldn't it be better to return an error saying this payment has already been inserted?

This is a general comment.

Reply:
Serial begins in 1 o 0 is the nil value for it, meaning that we got passed a payment with invalid ID which on second thought should be an error, let me correct that

Reply:
I don't think there's a requirement for idempotency, and the flip side of that coin is that you may be doing a lot of unnecessary operations without realizing it. I think it would be better to error here so that the caller knows they shouldn't be inserting something they've already inserted.